Adding `unionml` from PyPI

sugatoray opened the pull request:
<!--
Thank you very much for putting in this recipe PR!

This repository is very active, so if you need help with a PR, please let the
right people know. There are language-specific teams for reviewing recipes.

| Language        | Name of review team           |
| --------------- | ----------------------------- |
| python          | `@conda-forge/help-python`    |
| python/c hybrid | `@conda-forge/help-python-c`  |
| r               | `@conda-forge/help-r`         |
| java            | `@conda-forge/help-java`      |
| nodejs          | `@conda-forge/help-nodejs`    |
| c/c++           | `@conda-forge/help-c-cpp`     |
| perl            | `@conda-forge/help-perl`      |
| Julia           | `@conda-forge/help-julia`     |
| ruby            | `@conda-forge/help-ruby`      |
| other           | `@conda-forge/staged-recipes` |

Once the PR is ready for review, please mention one of the teams above in a
new comment. i.e. `@conda-forge/help-some-language, ready for review!`
Then, a bot will label the PR as 'review-requested'.

Due to GitHub limitations, first time contributors to conda-forge are unable
to ping conda-forge teams directly, but you can [ask a bot to ping the team][1]
using a special command in a comment on the PR to get the attention of the
`staged-recipes` team. You can also consider asking on our [Gitter channel][2]
if your recipe isn't reviewed promptly.

[1]: https://conda-forge.org/docs/maintainer/infrastructure.html#conda-forge-admin-please-ping-team
[2]: https://gitter.im/conda-forge/conda-forge.github.io
-->

Checklist
- [x] Title of this PR is meaningful: e.g. "Adding my_nifty_package", not "updated meta.yaml".
- [x] License file is packaged (see [here](https://github.com/conda-forge/staged-recipes/blob/5eddbd7fc9d1502169089da06c3688d9759be978/recipes/example/meta.yaml#L64-L73) for an example).
- [x] Source is from official source.
- [x] Package does not vendor other packages. (If a package uses the source of another package, they should be separate packages or the licenses of all packages need to be packaged).
- [x] If static libraries are linked in, the license of the static library is packaged.
- [x] Build number is 0.
- [x] A tarball (`url`) rather than a repo (e.g. `git_url`) is used in your recipe (see [here](https://conda-forge.org/docs/maintainer/adding_pkgs.html#build-from-tarballs-not-repos) for more details).
- [x] GitHub users listed in the maintainer section have posted a comment confirming they are willing to be listed there.
- [x] When in trouble, please check our [knowledge base documentation](https://conda-forge.org/docs/maintainer/knowledge_base.html) before pinging a team.

conda-forge-linter commented:
Hi! This is the friendly automated conda-forge-linting service.

I just wanted to let you know that I linted all conda-recipes in your PR (```recipes/unionml```) and found it was in an excellent condition.

conda-forge-linter commented:
Hi! This is the friendly automated conda-forge-linting service.

I wanted to let you know that I linted all conda-recipes in your PR (```recipes/flytekit```, ```recipes/mangum```, ```recipes/unionml```) and found some lint.

Here's what I've got...


For **recipes/flytekit**:

 * Feedstock with the same name exists in conda-forge.

For **recipes/flytekit**:

 * A conda package with same name (flytekit) already exists.

conda-forge-linter commented:
Hi! This is the friendly automated conda-forge-linting service.

I wanted to let you know that I linted all conda-recipes in your PR (```recipes/flyteidl```, ```recipes/flytekit```, ```recipes/mangum```, ```recipes/unionml```) and found some lint.

Here's what I've got...


For **recipes/flyteidl**:

 * Feedstock with the same name exists in conda-forge.

For **recipes/flytekit**:

 * Feedstock with the same name exists in conda-forge.

For **recipes/flytekit**:

 * A conda package with same name (flytekit) already exists.

sugatoray commented:
## TODO

:fire: The recipes for `flytekit` and `flyteidl` will be dropped from this PR before it is merged finally.

sugatoray commented:
## Reason for Build Failure: `2/N`

[(error)](https://dev.azure.com/conda-forge/feedstock-builds/_build/results?buildId=524693&view=logs&j=6f142865-96c3-535c-b7ea-873d86b887bd&t=22b0682d-ab9e-55d7-9c79-49f3c3ba4823&l=3494)

```sh
import: 'flytekit'
import: 'flytekit.bin'
+ pip check
flytekit 1.0.5 requires docker, which is not installed.
flytekit 1.0.5 requires grpcio-status, which is not installed.
Tests failed for flytekit-1.0.5-pyhd8ed1ab_0.tar.bz2 - moving package to /home/conda/staged-recipes/build_artifacts/broken
WARNING:conda_build.build:Tests failed for flytekit-1.0.5-pyhd8ed1ab_0.tar.bz2 - moving package to /home/conda/staged-recipes/build_artifacts/broken
TESTS FAILED: flytekit-1.0.5-pyhd8ed1ab_0.tar.bz2
```

**Resolution**

Updated run-requirements.

- added `docker-py`
- added `grpcio-status`

conda-forge-linter commented:
Hi! This is the friendly automated conda-forge-linting service.

I wanted to let you know that I linted all conda-recipes in your PR (```recipes/flytekit```, ```recipes/mangum```, ```recipes/unionml```) and found some lint.

Here's what I've got...


For **recipes/flytekit**:

 * Feedstock with the same name exists in conda-forge.

For **recipes/flytekit**:

 * A conda package with same name (flytekit) already exists.

sugatoray commented:
## Reason for Build Failure: `3/N`

[(error)](https://dev.azure.com/conda-forge/feedstock-builds/_build/results?buildId=524913&view=logs&j=6f142865-96c3-535c-b7ea-873d86b887bd&t=22b0682d-ab9e-55d7-9c79-49f3c3ba4823&l=5109)

```sh
import: 'unionml'
import: 'unionml'
+ pip check
unionml 0.1.2 requires sklearn, which is not installed.
Tests failed for unionml-0.1.2-pyhd8ed1ab_0.tar.bz2 - moving package to /home/conda/staged-recipes/build_artifacts/broken
WARNING:conda_build.build:Tests failed for unionml-0.1.2-pyhd8ed1ab_0.tar.bz2 - moving package to /home/conda/staged-recipes/build_artifacts/broken
TESTS FAILED: unionml-0.1.2-pyhd8ed1ab_0.tar.bz2
```

**Resolution**

Added `scikit-learn` to `test:requires`

sugatoray commented:
## Reason for Build Error: `4/N`

Possibly `git` is a necessary dependency of `unionml` that has not been explicitly added to unionml's list of dependencies. [(see here)](https://github.com/beeware/beeware/issues/32#issuecomment-588514845)

```sh
ImportError: Failed to initialize: Bad git executable.
```

For more details see below. [(error)]()

<details>
<summary>Click to expand</summary>
<p>

```sh
import: 'unionml'
import: 'unionml'
+ unionml --help
Traceback (most recent call last):
  File "/home/conda/staged-recipes/build_artifacts/unionml_1655850368352/_test_env_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehol/lib/python3.10/site-packages/git/__init__.py", line 87, in <module>
    refresh()
  File "/home/conda/staged-recipes/build_artifacts/unionml_1655850368352/_test_env_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehol/lib/python3.10/site-packages/git/__init__.py", line 76, in refresh
    if not Git.refresh(path=path):
  File "/home/conda/staged-recipes/build_artifacts/unionml_1655850368352/_test_env_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehol/lib/python3.10/site-packages/git/cmd.py", line 341, in refresh
    raise ImportError(err)
ImportError: Bad git executable.
The git executable must be specified in one of the following ways:
    - be included in your $PATH
    - be set via $GIT_PYTHON_GIT_EXECUTABLE
    - explicitly set via git.refresh()

All git commands will error until this is rectified.

This initial warning can be silenced or aggravated in the future by setting the
$GIT_PYTHON_REFRESH environment variable. Use one of the following values:
    - quiet|q|silence|s|none|n|0: for no warning or exception
    - warn|w|warning|1: for a printed warning
    - error|e|raise|r|2: for a raised exception

Example:
    export GIT_PYTHON_REFRESH=quiet


The above exception was the direct cause of the following exception:

Traceback (most recent call last):
  File "/home/conda/staged-recipes/build_artifacts/unionml_1655850368352/_test_env_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehol/bin/unionml", line 6, in <module>
    from unionml.cli import app
  File "/home/conda/staged-recipes/build_artifacts/unionml_1655850368352/_test_env_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehol/lib/python3.10/site-packages/unionml/cli.py", line 15, in <module>
    from unionml.remote import get_app_version, get_model, get_model_execution
  File "/home/conda/staged-recipes/build_artifacts/unionml_1655850368352/_test_env_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehol/lib/python3.10/site-packages/unionml/remote.py", line 8, in <module>
    import git
  File "/home/conda/staged-recipes/build_artifacts/unionml_1655850368352/_test_env_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehol/lib/python3.10/site-packages/git/__init__.py", line 89, in <module>
    raise ImportError('Failed to initialize: {0}'.format(exc)) from exc
ImportError: Failed to initialize: Bad git executable.
The git executable must be specified in one of the following ways:
    - be included in your $PATH
    - be set via $GIT_PYTHON_GIT_EXECUTABLE
    - explicitly set via git.refresh()

All git commands will error until this is rectified.

This initial warning can be silenced or aggravated in the future by setting the
$GIT_PYTHON_REFRESH environment variable. Use one of the following values:
    - quiet|q|silence|s|none|n|0: for no warning or exception
    - warn|w|warning|1: for a printed warning
    - error|e|raise|r|2: for a raised exception

Example:
    export GIT_PYTHON_REFRESH=quiet

Tests failed for unionml-0.1.2-pyhd8ed1ab_0.tar.bz2 - moving package to /home/conda/staged-recipes/build_artifacts/broken
WARNING:conda_build.build:Tests failed for unionml-0.1.2-pyhd8ed1ab_0.tar.bz2 - moving package to /home/conda/staged-recipes/build_artifacts/broken
TESTS FAILED: unionml-0.1.2-pyhd8ed1ab_0.tar.bz2
##[error]Bash exited with code '1'.

```

</p>
</details>

sugatoray commented:
## Reason for Build Failure: `5/N`

Observation: `import git` is used in `unionml/remote.py`. But `git` was not included in the run-requirements so far.

```sh
Traceback (most recent call last):
  File "/home/conda/staged-recipes/build_artifacts/unionml_1655889312114/_test_env_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehol/bin/unionml", line 6, in <module>
    from unionml.cli import app
  File "/home/conda/staged-recipes/build_artifacts/unionml_1655889312114/_test_env_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehol/lib/python3.10/site-packages/unionml/cli.py", line 15, in <module>
    from unionml.remote import get_app_version, get_model, get_model_execution
  File "/home/conda/staged-recipes/build_artifacts/unionml_1655889312114/_test_env_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehol/lib/python3.10/site-packages/unionml/remote.py", line 8, in <module>
    import git
  File "/home/conda/staged-recipes/build_artifacts/unionml_1655889312114/_test_env_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehol/lib/python3.10/site-packages/git/__init__.py", line 89, in <module>
    raise ImportError('Failed to initialize: {0}'.format(exc)) from exc
ImportError: Failed to initialize: Bad git executable.
The git executable must be specified in one of the following ways:
    - be included in your $PATH
    - be set via $GIT_PYTHON_GIT_EXECUTABLE
    - explicitly set via git.refresh()

All git commands will error until this is rectified.

This initial warning can be silenced or aggravated in the future by setting the
$GIT_PYTHON_REFRESH environment variable. Use one of the following values:
    - quiet|q|silence|s|none|n|0: for no warning or exception
    - warn|w|warning|1: for a printed warning
    - error|e|raise|r|2: for a raised exception

Example:
    export GIT_PYTHON_REFRESH=quiet

Tests failed for unionml-0.1.2-pyhd8ed1ab_0.tar.bz2 - moving package to /home/conda/staged-recipes/build_artifacts/broken
WARNING:conda_build.build:Tests failed for unionml-0.1.2-pyhd8ed1ab_0.tar.bz2 - moving package to /home/conda/staged-recipes/build_artifacts/broken
TESTS FAILED: unionml-0.1.2-pyhd8ed1ab_0.tar.bz2
```

conda-forge-linter commented:
Hi! This is the friendly automated conda-forge-linting service.

I just wanted to let you know that I linted all conda-recipes in your PR (```recipes/unionml```) and found it was in an excellent condition.

sugatoray commented:
PR is ready. Request for review/merging.

cc: @conda-forge/help-python

BastianZim commented on recipes/unionml/meta.yaml:
```diff
@@ -0,0 +1,88 @@
+  entry_points:
+    - unionml = unionml.cli:app
+  script:
+    ## Ensure inclusion of requirements.txt
```
Is this really needed?

sugatoray replied:
I pushed a PR in upstream to include the requirements.txt and it was merged. So, I would request you to merge it as is for now. Once the next release of unionml is available on PyPI, I will update the recipe and remove the workaround.

- unionml PR: https://github.com/unionai-oss/unionml/pull/125

thewchan replied:
I didn't know this was being written and did my own recipe; you shouldn't need this script here. My recipe worked without it.

thewchan replied:
CF: https://github.com/conda-forge/staged-recipes/pull/19414

sugatoray replied:
@thewchan Thank you, for the suggestion. Removed the workaround.

sugatoray commented:
I believe, the recipe is ready now. PLease let me know if you have any other concerns.

cc: @BastianZim @thewchan
